core/winnt/src/TWinNTSystem.cxx

```diff
@@ -2093,18 +2109,17 @@ void *TWinNTSystem::OpenDirectory(const char *fdir)
          entry[strlen(dir)-1] = '\0';
       strlcat(entry,"*",nche);
 
-      HANDLE searchFile;
-      searchFile = ::FindFirstFile(entry, &fFindFileData);
-      if (searchFile == INVALID_HANDLE_VALUE) {
+      dirp->fSearchFile = ::FindFirstFile(entry, &dirp->fFindFileData);
+      if (dirp->fSearchFile == INVALID_HANDLE_VALUE) {
          ((TWinNTSystem *)gSystem)->Error( "Unable to find' for reading:", entry);
          delete [] entry;
          delete [] dir;
          return nullptr;
       }
       delete [] entry;
       delete [] dir;
-      fFirstFile = kTRUE;
-      return searchFile;
+      dirp->fFirstFile = kTRUE;
+      return dirp;
    }
 
    delete [] entry;
```

Comment:
Doesn't this leak `dir`?

Reply:
Look carefully
```
      delete [] entry;
      delete [] dir;
      dirp->fFirstFile = kTRUE;
      return dirp;
```

Reply:
... And indeed also there I missed it. Sorry!